### ISSUE TITLE
genesis bridge: connect to fraud

### ISSUE THREAD

**Issue by danwt** [dymensionxyz/dymension#930]
When there is a protocol violation because the rollapp did not adhere to the dymension rollapp standard, then actually punish the chain

https://github.com/dymensionxyz/dymension/blob/738ba59f18e08586c01326fffa8c52584a14d9f5/x/rollapp/transfergenesis/ibc_module.go#L208-L215

**Comment by mtsitrin** [comment]
due to the changes introduced with the `IRO` module,
the logic will be:
* No IRO - rollapp's transfers enabled by default. genesisTransfer is skipped
* With IRO - genesisTransfer is validated before IRO settlement and transfer enabling.

If genesis transfer is required (IRO exists), but fails validation, it will be "ignored", but won't freeze the rollapp.
As we want to give the rollapp opportunity to fix it's state post-genesis

bottom line, genesisTransfer won't trigger fraud handler anymore
